Add setting for the volume of ruleset elements (hitsounds)

leocb opened the pull request:
Hi! First time contributing here!

This PR adds the option to mute only hitsounds.

The setting is available on the main menu and on the play "quick" menu

I followed what was already in place for the mute mod and, while this aims to only mute hitsounds, AFAIK my implementation will mute any sounds coming from the ruleset drawable, I don't know if this impacts other desired sounds, if this is acceptable or if the name should be changed from "hitsounds" to something else to better reflect this "ruleset mute".

leocb committed:
Add settings option to mute only hitsounds

peppy commented:
Thanks for your contribution.

I'd say the more desired behaviour is to have volume control for hitsounds, rather than a toggle.

leocb commented:
This is a toggle as it's being applied as a multiplier on top of the effects volume, having a slider would make 100% the same as the current effects slider, which may be confusing.
I don't know if it can be easily separated into another slider without messing with framework too.

bdach commented:
Also against this feature in its current form.

> I don't know if it can be easily separated into another slider without messing with framework too.

Easiest path towards that would probably be to put gameplay samples on a separate mixer. Which is something we may want to do at some point anyway, but definitely will not be a 10-line diff like this is.

peppy commented:
> Easiest path towards that would probably be to put gameplay samples on a separate mixer. Which is something we may want to do at some point anyway, but definitely will not be a 20-line diff like this is.

Failing this, exposing an audio adjustment for the gameplay audio container should be relatively simple.

leocb committed:
Merge branch 'ppy:master' into master

leocb committed:
Adjustable gameplay volume slider

Add Volume slider for gameplay sounds
Modify ruleset container to AudioMixerContainer
Swap Music and effects sliders

leocb commented:
I made some modifications in framework to allow mixers to have their own volume, now this PR depends on this on: ppy/osu-framework#6326
I removed the option from the play "quick" settings
The ruleset container was changed to AudioMixerContainer and now has it's own mixer separate from the UI sfx
I added a slider in the settings for the gameplay sounds volume and called it `Gameplay` feel free to suggest other names that better reflect what this does

fadedlamp42 commented:
Seems like this went a bit cold, but think that while we're at it you could make a "constant volume" tick to satisfy https://github.com/ppy/osu/discussions/18654 as well?

frenzibyte commented:
This is not relevant in any way to this PR. It should be done separately.

Erujuru commented:
Hello everyone!
I hope you are all well. I'm not technically skilled to be able to help with this issue, so I am definitely not here to complain or make any sort of negative comment. Just here to bump the feature discussion since it seems to have gone cold for a few months. I made a similar feature request bump awhile back here: https://osu.ppy.sh/community/forums/topics/1887559?n=4

All the best!
